hardware/arduino/avr/cores/arduino/USBCore.cpp

```diff
@@ -273,12 +271,15 @@ int USB_Send(u8 ep, const void* d, int len)
 	}
 
 	int r = len;
+
+	bool sendZlp = (len % USB_EP_SIZE) == 0;
```

Comment:
I think that this is the wrong way to handle ZLP since the buffer could be filled in multiple rounds, for example this sketch still fails after the patch:

```C++
void setup() { }
void loop() {
  if (Serial) {
    if (Serial.available() > 0) {
      Serial.read();
      char buff[65] = "<--3456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABC-->";
      Serial.write(buff,32);
      Serial.write(buff+32,32);
      Serial.flush();
    }
  }
}
```